File: app/javascript/graphql/mutations/sendInvitationMail.ts
import { gql } from "@apollo/client";

export const SEND_INVITATION_MAIL = gql`
  mutation SendInvitationMail($email: String!, $teamId: ID!) {
    sendInvitationMail(input: { email: $email, teamId: $teamId }) {
      message
    }
  }
`;

File: app/graphql/mutations/send_invitation_mail.rb
module Mutations
  class SendInvitationMail < Mutations::BaseMutation
    field :message, String, null: false

    argument :email, String, required: true
    argument :team_id, ID, required: true

    def resolve(email:, team_id:)
      team = current_user.created_teams.find(team_id)
      invitation = team.invitaitons.create!(email: email, token: Nanoid.generate)
      InviteMailer.invite_to_team(invitation).deliver_now
      { message: "招待メールを送信しました。" }
    rescue StandardError => e
      GraphQL::ExecutionError.new(e.message)
    end
  end
end
